### PR TITLE
Code for Week 7 (Post-Testing)

### PR DESCRIPTION
**Pull request by connernilsen**
This PR is waiting on #7

### CODE REVIEW

File: service/src/test/java/com/codeforcommunity/processor/PostsProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+        assertTrue(comment.getClapCount() >= 0);
+        assertNotNull(comment.getDateCreated());
+        assertFalse(comment.getDateCreated().isBlank());
+      }
+    }
+  }
+
+  // Write a test which runs multiple times with different inputs.
+  @ParameterizedTest
+  @ValueSource(ints = {1, 2, 5, 14})
```
Comment (liammoyn) [inline service/src/test/java/com/codeforcommunity/processor/PostsProcessorTest.java:113]: Do you describe what these annotations are anywhere?

File: api/src/test/java/com.codeforcommunity/rest/PostsRouterTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    this.vertxRouter = mock(Router.class);
+    this.route = mock(Route.class);
+    this.ctx = mock(RoutingContext.class);
+
+    // Mock for the IRouter#end method. We don't have to handle HttpServerResponse#end method, since
+    // unmocked methods just return null.
+    res = mock(HttpServerResponse.class);
+    when(res.setStatusCode(anyInt())).thenReturn(res);
+    when(res.putHeader(anyString(), anyString())).thenReturn(res);
+    when(ctx.response()).thenReturn(res);
```
Comment (maxsebso) [inline api/src/test/java/com.codeforcommunity/rest/PostsRouterTest.java]: maybe a little annotation if you think it is necessary rather then in a big paragraph

**Review by maxsebso — APPROVED** [review]
You done good. Like all the explanations

File: service/src/test/java/com/codeforcommunity/processor/PostsProcessorTest.java (hunk 1 above)
Comment (kiminlee2000225) [reply service/src/test/java/com/codeforcommunity/processor/PostsProcessorTest.java:113]: ++ on that. I think it'll be nice to explain what `@ValueSource(ints = {-1, STUB_POST_COUNT + 1})` mean as well!

File: service/src/test/java/com/codeforcommunity/processor/PostsProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+    processor.clapPost(1);
+    // Check that the value changed.
+    assertEquals(clapCount + 1, post.getClapCount());
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {-1, STUB_POST_COUNT + 1})
+  public void testClapInvalidPost(int postId) {
+    try {
+      // Try to add a comment to an invalid post.
```
Comment (kiminlee2000225) [inline service/src/test/java/com/codeforcommunity/processor/PostsProcessorTest.java]: is this supposed to be "Try to add a clap to an invalid post." instead?

File: api/src/main/java/com/codeforcommunity/rest/PostsRouter.java [hunk 4]
```diff
@@ -20,14 +20,43 @@
...
     this.processor = postsProcessor;
   }
 
   @Override
   public Router initializeRouter(Vertx vertx) {
-    // Set a router object
-    Router router = Router.router(vertx);
+    // Set a router object. We'll now be getting this from the externs so that we can test using a
+    // mocked router.
+    Router router = externs.getRouter(vertx);
```
Comment (jtavera235) [inline api/src/main/java/com/codeforcommunity/rest/PostsRouter.java]: Just to say consistent would this be this.extern.getRouter(vertx);?

**Review by jtavera235 — APPROVED** [review]

File: service/src/test/java/com/codeforcommunity/MockitoTestExamples.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    Map<Integer, PostRecord> postMap = postTable.getUnderlyingDb();
+    // Remember that the IDs are 0-indexed.
+    PostRecord recentPost = postMap.get(postMap.size() - 1);
+    assertEquals(author, recentPost.getAuthor());
+    assertEquals(title, recentPost.getTitle());
+    assertEquals(body, recentPost.getBody());
+    // Make sure the count of posts has increased by 1.
+    assertEquals(STUB_POST_COUNT + 1, postMap.size());
+
+    //
```
Comment (liammoyn) [inline service/src/test/java/com/codeforcommunity/MockitoTestExamples.java]: Missing comment?

**Review by liammoyn — APPROVED** [review]

**Pull request merged**